### PR TITLE
fix: broken link in python and also add another similar resource

### PR DESCRIPTION
**Pull request by dev-aly3n**
This PR fixes issue #5043 
change this [link](https://www.section.io/engineering-education/implementing-binary-search-tree-using-python/) to [archive version](https://web.archive.org/web/20230601181553/https://www.section.io/engineering-education/implementing-binary-search-tree-using-python/) as the site is unavailable and redirected to host provider.

also added another similar resource: [https://blog.boot.dev/computer-science/binary-search-tree-in-python/](https://blog.boot.dev/computer-science/binary-search-tree-in-python/)

### CODE REVIEW

**Comment by bcdady** [comment]
And/or
<https://www.pythonforbeginners.com/data-structures/binary-search-tree-in-python>

**Comment by dev-aly3n** [comment]
Thanks, @bcdady 
I also added the resource you mentioned

**Comment by dev-aly3n** [comment]
Hi @arikchakma , Is there any chance you could kindly take a look at these PRs?:

#5061 , #5014 , #5013

File: src/data/roadmaps/python/content/101-data-structures-and-algorithms/103-binary-search-trees.md [hunk 1]
```diff
@@ -5,5 +5,7 @@ A binary search tree, also called an ordered or sorted binary tree, is a rooted
 Visit the following resources to learn more:
 
 - [Tree Data Structure | Illustrated Data Structures](https://www.youtube.com/watch?v=S2W3SXGPVyU)
-- [How to Implement Binary Search Tree in Python](https://www.section.io/engineering-education/implementing-binary-search-tree-using-python/)
+- [WRITING A BINARY SEARCH TREE IN PYTHON WITH EXAMPLES](https://blog.boot.dev/computer-science/binary-search-tree-in-python/)
```
Comment (arikchakma) [inline src/data/roadmaps/python/content/101-data-structures-and-algorithms/103-binary-search-trees.md:8]:
```suggestion
- [Writing a Binary Search Tree in Python With Examples](https://blog.boot.dev/computer-science/binary-search-tree-in-python/)
```

**Review by arikchakma — APPROVED** [review]

**Pull request merged**